sys/kern/src/arch/arm_m.rs

```diff
@@ -73,6 +73,7 @@
 use core::arch::{self, global_asm};
 use core::sync::atomic::{AtomicBool, AtomicPtr, AtomicU32, Ordering};
 
+use serde::{Deserialize, Serialize};
```

Comment:
It feels a little weird to be bringing these into the kernel, mostly because those seem much higher level. I don't have a better idea.

Reply:
Let me remind myself where this is actually used nowadays, maybe we can remove it.

Reply:
It seems it's required here because the (cross-platform) `RegionDesc` in `kern::descs` derives `Serialize` and `Deserialize`, and the `RegionDescExt` struct must also to not break that. I don't actually see any obvious places where we serialize or deserialize `RegionDesc`s --- I wonder if this is a historical artifact. Did we once use these types in the build code when deserializing an application's config file? It looks like we now use a different type in `build_kconfig`: https://github.com/oxidecomputer/hubris/blob/874c3b899af5833a501580bd5ad10f8acb3fe3d0/build/kconfig/src/lib.rs#L96-L118

Reply:
Good news! I can remove it, and will push the commit shortly.